### PR TITLE
Add subcodes to kuzzle core entrypoints

### PR DESCRIPTION
**Pull request by Yoann-Abbes**
## What does this PR do ?

Jira : [Subtask](https://jira.kaliop.net/browse/KZL-1200) from this [story](https://jira.kaliop.net/browse/KZL-1192)

This PR is to merge into the `KZL-1035-subcode-error` [(here)](https://github.com/kuzzleio/kuzzle/pull/1306) branch, such as the PRs to come.

Add subcodes to : 
`authController`
`adminController`
`bulkController`
`securityController`
`memoryStorage`
`funnelController`
`collectionController`
`indexController`
`documentController`

Add/adapt unit test.

Did not do for `PluginImplementationError`, it will be in another PR (it's another [subtask](https://jira.kaliop.net/browse/KZL-1199))

### How should this be manually tested?

unit tests

### Boyscout

80 columns

### CODE REVIEW

File: lib/api/controllers/adminController.js [hunk 1]
```diff
@@ -93,7 +90,7 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
     cacheEngine.flushdb();
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR TypeError can be thrown as "cacheEngine" might be null or undefined here. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)

File: lib/api/controllers/authController.js [hunk 2]
```diff
@@ -101,7 +100,8 @@ class AuthController extends BaseController {
     return this.kuzzle.repositories.token.expire(request.context.token)
       .then(() => ({acknowledged: true}))
       .catch(err => {
-        const error = new KuzzleInternalError('Error while forcing token expiration.');
+        //const error = new KuzzleInternalError('Error while forcing token expiration.');
```
Comment (kuzzle) [inline lib/api/controllers/authController.js]: MAJOR Remove this commented out code. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ACommentedCode)

**Comment by kuzzle** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: lib/api/controllers/adminController.js [hunk 3]
```diff
@@ -93,10 +90,12 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (cacheEngine !== null && cacheEngine !== undefined) {
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR Change this condition so that it does not always evaluate to "true". [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2583)
MAJOR Remove this "!==" check; it will always be true. Did you mean to use "!="? [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3403)

File: lib/api/controllers/adminController.js [hunk 4]
```diff
@@ -93,10 +90,12 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (! (cacheEngine === null || cacheEngine === undefined)) {
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR Change this condition so that it does not always evaluate to "false". [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2583)
MAJOR Remove this "===" check; it will always be false. Did you mean to use "=="? [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3403)

File: lib/api/controllers/adminController.js [hunk 5]
```diff
@@ -93,10 +90,12 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (! (cacheEngine == null || cacheEngine == undefined)) {
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR Replace "==" with "===". [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AEqEqEq)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: lib/api/controllers/adminController.js [hunk 6]
```diff
@@ -93,10 +90,12 @@ class AdminController extends BaseController {
       cacheEngine = this.kuzzle.services.list[database];
 
     if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (! (!cacheEngine || cacheEngine === undefined)) {
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR Remove this "===" check; it will always be false. Did you mean to use "=="? [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3403)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: lib/config/error-codes/api.json [hunk 7]
```diff
@@ -1,6 +1,16 @@
 {
   "code": 1,
   "subdomains": {
+    "base": {
+      "code": 1,
+      "errors": {
+        "invalid_value_type": {
```
Comment (scottinet) [inline lib/config/error-codes/api.json]: I expect a generic error with that code, not one regarding only booleans. Change either the code or the message.

File: lib/api/controllers/adminController.js [hunk 8]
```diff
@@ -218,7 +217,7 @@ class AdminController extends BaseController {
     const action = request.input.action;
 
     if (_locks[action]) {
-      throw new PreconditionError(message);
+      this.throw('precondition', message);
```
Comment (scottinet) [inline lib/api/controllers/adminController.js]: Wrong condition name: should be something like `action_locked` or something.
As it is, reading the error configuration file makes it impossible to understand what this error does.

File: lib/config/error-codes/api.json [hunk 9]
```diff
@@ -130,6 +145,181 @@
...
+        },
+        "user_not_found": {
+          "code": 3,
+          "message": "User with id %s not found",
+          "class": "NotFoundError"
+        },
+        "admin_exists": {
+          "code": 4,
+          "message": "Admin user is already set",
+          "class": "KuzzleError"
```
Comment (scottinet) [inline lib/config/error-codes/api.json]: should be a PreconditionError (the error currently thrown by Kuzzle is wrong, this is not an internal problem)

File: lib/config/error-codes/api.json [hunk 10]
```diff
@@ -130,6 +145,181 @@
...
+          "class": "BadRequestError"
+        },
+        "cant_update_creds": {
+          "code": 6,
+          "message": "Cannot update credentials: unknown kuid %s",
+          "class": "BadRequestError"
+        },
+        "delete_limit_reached": {
+          "code": 7,
+          "message": "Number of delete to perform exceeds the server configured value %s",
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
          "message": "The number of deletes to perform exceeds the server configured value %s",
```

File: lib/config/error-codes/api.json [hunk 11]
```diff
@@ -66,25 +81,25 @@
       "errors": {
         "token_force_expire": {
           "code": 1,
-          "message": "Error while forcing token expiration",
+          "message": "Error while forcing token expiration.",
```
Comment (scottinet) [inline lib/config/error-codes/api.json]: Why not, but in that case you have to be consistent and apply that modification for all messages.

File: lib/config/error-codes/api.json [hunk 12]
```diff
@@ -130,6 +145,181 @@
...
+          "class": "BadRequestError"
+        },
+        "aggregate_parameter": {
+          "code": 6,
+          "message": "Invalid aggregate parameter",
+          "class": "BadRequestError"
+        },
+        "float_expected": {
+          "code": 7,
+          "message": "Invalid value for parameter %s : float expected.",
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
          "message": "Invalid value for parameter %s: float expected.",
```

File: lib/config/error-codes/api.json [hunk 13]
```diff
@@ -130,6 +145,181 @@
...
+          "class": "BadRequestError"
+        },
+        "integer_expected": {
+          "code": 8,
+          "message": "Invalid value for parameter %s : float expected.",
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
          "message": "Invalid value for parameter %s: integer expected.",
```

File: lib/config/error-codes/api.json [hunk 14]
```diff
@@ -66,37 +81,37 @@
...
+          "message": "Invalid token.",
           "class": "UnauthorizedError"
         }
       }
     },
     "collection": {
       "code": 5,
       "errors": {
-        "index_not_exist": {
+        "index_not_exists": {
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
        "index_does_not_exist": {
```

File: lib/config/error-codes/api.json [hunk 15]
```diff
@@ -66,37 +81,37 @@
...
           "code": 1,
-          "message": "The index %s does not exist",
+          "message": "The index %s does not exist.",
           "class": "PreconditionError"
         },
-        "collection_not_exist": {
+        "collection_not_exists": {
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
        "collection_does_not_exist": {
```

File: lib/config/error-codes/api.json [hunk 16]
```diff
@@ -106,27 +121,202 @@
...
+          "code": 9,
+          "message": "At least 1 source key must be provided.",
+          "class": "BadRequestError"
+        },
+        "no_score_member_pair": {
+          "code": 10,
+          "message": "At least 1 score/member pair must be provided.",
+          "class": "BadRequestError"
+        },
+        "too_much_score_member_pair": {
```
Comment (scottinet) [inline lib/config/error-codes/api.json]:
```suggestion
        "too_many_score_member_pairs": {
```

File: test/api/controllers/adminController.test.js [hunk 17]
```diff
@@ -52,15 +48,16 @@ describe('Test: admin controller', () => {
...
     });
 
-    it('should raise an error if database does not exist', done => {
+    it('should raise an error if database does not exist', () => {
       request.input.args.database = 'city17';
+      adminController.throw = sinon.spy();
 
       try {
         adminController.resetCache(request);
-        done(new Error('Should not resolves'));
```
Comment (scottinet) [inline test/api/controllers/adminController.test.js]: I disagree with this removal (albeit the typo should be fixed): without it, this test passes if the admin controller does not throw.

File: test/api/controllers/adminController.test.js [hunk 18]
```diff
@@ -208,10 +205,15 @@ describe('Test: admin controller', () => {
     it('should throw an error if shutdown is in progress', () => {
       AdminController.__set__('_locks', { shutdown: true });
       adminController = new AdminController(kuzzle);
+      adminController.throw = sinon.spy();
 
-      return should(() => {
+      try {
```
Comment (scottinet) [inline test/api/controllers/adminController.test.js]: This should be reverted: the test passes if the function doesn't throw.

File: test/api/controllers/authController.test.js [hunk 19]
```diff
@@ -286,14 +286,21 @@ describe('Test the auth controller', () => {
...
-        {
-          token: {userId: 'foo', _id: 'bar', refreshed: true},
-          user: {_id: 'bar'}
-        }
-      )))
-        .throw(UnauthorizedError, {message: 'Invalid token'});
+      authController.throw = sinon.spy();
+
+      try {
+        authController.refreshToken(new Request(
```
Comment (scottinet) [inline test/api/controllers/authController.test.js]: This test passes if the function doesn't throw.

File: test/api/controllers/collectionController.test.js [hunk 20]
```diff
@@ -112,21 +111,27 @@ describe('Test: collection controller', () => {
 
     it('should return a dedicated error if the index does not exist', () => {
       kuzzle.indexCache.exists.resolves(false);
+      collectionController.throw = sinon.spy();
 
-      return should(collectionController.getSpecifications(request))
-        .rejectedWith(
-          PreconditionError,
-          {message: `The index '${index}' does not exist`});
+      return collectionController.getSpecifications(request)
```
Comment (scottinet) [inline test/api/controllers/collectionController.test.js]: This test does nothing if the function doesn't neither throw nor reject.

File: test/api/controllers/collectionController.test.js [hunk 21]
```diff
@@ -112,21 +111,27 @@ describe('Test: collection controller', () => {
...
     it('should return a dedicated error if the collection does not exist', () => {
       kuzzle.indexCache.exists.onFirstCall().resolves(true);
       kuzzle.indexCache.exists.onSecondCall().resolves(false);
+      collectionController.throw = sinon.spy();
 
-      return should(collectionController.getSpecifications(request))
-        .rejectedWith(
-          PreconditionError,
-          {message: `The collection '${collection}' does not exist`});
+      return collectionController.getSpecifications(request)
```
Comment (scottinet) [inline test/api/controllers/collectionController.test.js]: This test does nothing if the function doesn't neither throw nor reject.

File: test/api/controllers/collectionController.test.js [hunk 22]
```diff
@@ -146,9 +151,15 @@ describe('Test: collection controller', () => {
       request.input.args.from = 0;
       request.input.args.size = 20;
       request.input.action = 'searchSpecifications';
+      collectionController.throw = sinon.spy();
 
-      return should(() => collectionController.searchSpecifications(request))
-        .throw('Search page size exceeds server configured documents limit (1)');
+      try {
+        collectionController.searchSpecifications(request);
```
Comment (scottinet) [inline test/api/controllers/collectionController.test.js]: This test does nothing if the function doesn't neither throw nor reject.

File: test/api/controllers/collectionController.test.js [hunk 23]
```diff
@@ -588,10 +599,15 @@ describe('Test: collection controller', () => {
 
     it('should reject the request if an invalid "type" argument is provided', () => {
       request = new Request({index: 'index', type: 'foo'});
+      collectionController.throw = sinon.spy();
 
-      return should(() => {
+      try {
```
Comment (scottinet) [inline test/api/controllers/collectionController.test.js]: This test does nothing if the function doesn't neither throw nor reject.

File: lib/api/controllers/funnelController.js [hunk 24]
```diff
@@ -36,9 +36,8 @@ const
   AdminController = require('./adminController'),
   Request = require('kuzzle-common-objects').Request,
   {
-    BadRequestError,
-    ForbiddenError,
     KuzzleError,
+    errorsManager = require('../../config/error-codes/throw.js');
```
Comment (kuzzle) [inline lib/api/controllers/funnelController.js]: MAJOR Parse error at line 40 column 65:

30:   DocumentController = require('./documentController'),
31:   IndexController = require('./indexController'),
32:   MemoryStorageController = require('./memoryStorageController'),
33:   RealtimeController = require('./realtimeController'),
34:   SecurityController = require('./securityController'),
35:   ServerController = require('./serverController'),
36:   AdminController = require('./adminController'),
37:   Request = require('kuzzle-common-objects').Request,
38:   {
39:     KuzzleError,
40:     errorsManager = require('../../config/error-codes/throw.js');
                                                                    ^
41:     PluginImplementationError,
42:     ServiceUnavailableError,
43:     UnauthorizedError
44:   } = require('kuzzle-common-objects').errors,
45:   documentEventAliases = require('../../config/documentEventAliases'),
46:   DocumentExtractor = require('../../util/DocumentExtractor');
47: 
48: /**
49:  * @class CacheItem
50:  */ [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AParsingError)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: test/api/controllers/adminController.test.js [hunk 25]
```diff
@@ -54,13 +50,13 @@ describe('Test: admin controller', () => {
...
+      adminController.throw = sinon.spy();
 
       try {
         adminController.resetCache(request);
-        done(new Error('Should not resolves'));
+        done(adminController.throw('database_not_found', 'city17'));
       } catch (e) {
-        should(e).be.instanceOf(BadRequestError);
-        done();
+        done(e);
```
Comment (benoitvidis) [inline test/api/controllers/adminController.test.js]: This test always passes.

File: test/api/controllers/adminController.test.js [hunk 26]
```diff
@@ -205,13 +201,17 @@ describe('Test: admin controller', () => {
...
+      adminController.throw = sinon.spy();
 
-      return should(() => {
+      try {
         adminController.shutdown(request);
-      }).throw(PreconditionError);
+        done(adminController.throw('precondition', 'Kuzzle is already shutting down.'));
+      } catch (e) {
+        done(e);
+      }
```
Comment (benoitvidis) [inline test/api/controllers/adminController.test.js]: This test always passes

File: test/api/controllers/authController.test.js [hunk 27]
```diff
@@ -285,15 +284,21 @@ describe('Test the auth controller', () => {
...
+          {},
+          {
+            token: { userId: 'foo', _id: 'bar', refreshed: true },
+            user: { _id: 'bar' }
+          }
+        ));
+        done(authController.throw('invalid_token'));
+      } catch (e) {
+        done(e);
+      }
```
Comment (benoitvidis) [inline test/api/controllers/authController.test.js]: This test always passes.

File: test/api/controllers/collectionController.test.js [hunk 28]
```diff
@@ -112,43 +112,44 @@ describe('Test: collection controller', () => {
...
     it('should return a dedicated error if the index does not exist', () => {
       kuzzle.indexCache.exists.resolves(false);
-
-      return should(collectionController.getSpecifications(request))
-        .rejectedWith(
-          PreconditionError,
-          {message: `The index '${index}' does not exist`});
+      
+      return collectionController.getSpecifications(request)
+        .should.be.rejectedWith(PreconditionError, {message: `The index ${index} does not exist.`});
```
Comment (benoitvidis) [inline test/api/controllers/collectionController.test.js]: Here and for all other occurences: prefer `should(` over its `Object` override form.

File: test/api/controllers/collectionController.test.js [hunk 29]
```diff
@@ -112,43 +112,44 @@ describe('Test: collection controller', () => {
...
 
-      return should(() => collectionController.searchSpecifications(request))
-        .throw('Search page size exceeds server configured documents limit (1)');
+      try {
+        collectionController.searchSpecifications(request);
+        done(collectionController.throw('search_page_size', 1));
+      } catch (e) {
+        done(e);
+      }
     });
```
Comment (benoitvidis) [inline test/api/controllers/collectionController.test.js]: The test always passes

File: test/api/controllers/collectionController.test.js [hunk 30]
```diff
@@ -586,12 +587,16 @@ describe('Test: collection controller', () => {
...
+      collectionController.throw = sinon.spy();
 
-      return should(() => {
+      try {
         collectionController.list(request);
-      }).throw(BadRequestError);
+        done(collectionController.throw('invalid_type_argument', 'foo'));
+      } catch (e) {
+        done(e);
+      }
```
Comment (benoitvidis) [inline test/api/controllers/collectionController.test.js]: This test always passes.

File: lib/api/controllers/adminController.js [hunk 31]
```diff
@@ -92,10 +89,10 @@ class AdminController extends BaseController {
...
       cacheEngine = this.kuzzle.services.list[database];
 
-    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+    if (! (cacheEngine !== undefined &&
+        typeof cacheEngine.flushdb === 'function')) {
+      this.throw('database_not_found', database);
     }
-
     cacheEngine.flushdb();
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR TypeError can be thrown as "cacheEngine" might be null or undefined here. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: lib/api/controllers/adminController.js [hunk 32]
```diff
@@ -92,11 +89,14 @@ class AdminController extends BaseController {
...
 
-    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+    if (! (cacheEngine !== undefined &&
+        typeof cacheEngine.flushdb === 'function')) {
+      this.throw('database_not_found', database);
     }
 
-    cacheEngine.flushdb();
+    if (cacheEngine) {
```
[the remark this answers is not in the document]
Comment (Yoann-Abbes) [reply lib/api/controllers/adminController.js]: Sonarqube forced me to check

**Comment by Yoann-Abbes** [comment]
> There are many more errors in unit tests that must be fixed: many unit tests verifying if exception are thrown have been modified in such a way that they pass if the tested function does not throw anymore.

It should be corrected now

File: test/api/controllers/collectionController.test.js (hunk 28 above)
Comment (Yoann-Abbes) [reply test/api/controllers/collectionController.test.js]: Okay
but why ?

File: test/api/controllers/authController.test.js (hunk 27 above)
Comment (benoitvidis) [reply test/api/controllers/authController.test.js]: Still not good (here and other cases)
The test will pass in both cases where `refreshToken` throws or not. 
Actually, here, as `throw` is overriden to a `spy`, the `catch` clause is never executed.

possible implementation:
```javascript 
    it('should throw if the token has already been refreshed', () => {
      authController.throw = sinon.spy();

      authController.refreshToken(new Request(
        {},
        {
          token: { userId: 'foo', _id: 'bar', refreshed: true },
          user: { _id: 'bar' }
        }
      ));
      should(authController.throw).be.calledWith('invalid_token');
    });
```
Or, by actually throwing:

```javascript
it('should throw', () => {
  return should(authController.refreshToken(request))
    .be.rejectedWith(ExpectedError, {message: 'some error'});
});
```

File: test/api/controllers/collectionController.test.js (hunk 28 above)
Comment (Aschen) [reply test/api/controllers/collectionController.test.js]: For consistency with the other tests

File: lib/api/controllers/adminController.js [hunk 33]
```diff
@@ -92,20 +89,24 @@ class AdminController extends BaseController {
...
       cacheEngine = this.kuzzle.services.list[database];
 
-    if (! (cacheEngine !== undefined && typeof cacheEngine.flushdb === 'function')) {
-      throw new BadRequestError(`Database ${database} not found`);
+    if (! (cacheEngine !== undefined &&
+        typeof cacheEngine.flushdb === 'function')) {
+      this.throw('database_not_found', database);
     }
 
     cacheEngine.flushdb();
```
Comment (kuzzle) [inline lib/api/controllers/adminController.js]: MAJOR TypeError can be thrown as "cacheEngine" might be null or undefined here. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2259)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: lib/api/controllers/adminController.js (hunk 32 above)
[the remark this answers is not in the document]
Comment (Yoann-Abbes) [reply lib/api/controllers/adminController.js]: I re-tested and Sonarqube keeps raising an error :/
Advised by @scottinet I changed the way to solve it

**Comment by scottinet** [comment]
There is still this comment from @benoitvidis that is left unresolved: https://github.com/kuzzleio/kuzzle/pull/1333/files#r306856634

Also, I don't understand why every single files you changes has been entirely (badly) reformated. Did you use some kind of automated code formatting tool (such as prettier)?

**Comment by Yoann-Abbes** [comment]
> There is still this comment from @benoitvidis that is left unresolved: https://github.com/kuzzleio/kuzzle/pull/1333/files#r306856634
> 
> Also, I don't understand why every single file you changes has been entirely (badly) reformated. Did you use some kind of automated code formatting tool (such as prettier)?

I rechecked the files, i reversed the bad changes and just kept the 80 column wrapping
For the comment i missed, its ok now.

File: test/api/core/hotelClerk/addSubscription.test.js [hunk 34]
```diff
@@ -206,7 +206,6 @@ describe('Test: hotelClerk.addSubscription', () => {
       id: 'foobar',
     }));
     return hotelClerk.addSubscription(request)
-      .then(() => {throw new Error('should not happen');})
```
Comment (scottinet) [inline test/api/core/hotelClerk/addSubscription.test.js]: This should not be removed: if `hotelClerk.addSubscription` resolves instead of rejecting with an error, the test incorrectly passes.

File: test/mocha.opts [hunk 35]
```diff
@@ -1,6 +1,5 @@
 --bail
 --require should-sinon
---reporter dot
```
Comment (scottinet) [inline test/mocha.opts]: why did you change the reporter?

Comment (Yoann-Abbes) [reply test/mocha.opts]: To have it more readable
Why is it better to have dots ?

File: test/api/core/hotelClerk/addSubscription.test.js (hunk 34 above)
Comment (Yoann-Abbes) [reply test/api/core/hotelClerk/addSubscription.test.js]: Hmm, its a mistake, i cant see why i did that

File: lib/api/controllers/routerController.js [hunk 36]
```diff
@@ -23,7 +23,8 @@
 
 const
   HttpRouter = require('../core/httpRouter'),
-  PluginImplementationError = require('kuzzle-common-objects').errors.PluginImplementationError,
+  PluginImplementationError =
```
[the remark this answers is not in the document]
Comment (Yoann-Abbes) [reply lib/api/controllers/routerController.js]: will be done by merging 1-dev

File: test/mocha.opts (hunk 35 above)
Comment (scottinet) [reply test/mocha.opts]: We changed the reporter a while back because the `dot` reporter has the following advantages:

* the dot reporter filters passing tests out, making errors far more visible
* the default reporter makes it almost impossible to spot warning (such as unhandled promise rejection warnings) or residual console.log outputs amongst the huge wall of text

All in all, it makes the parsing of travis logs much more manageable.

Also, before making this kind of changes, we have the habit of asking if it suits everyone on slack.

Comment (Yoann-Abbes) [reply test/mocha.opts]: Okay i see now, thanks for the explaination

File: lib/api/controllers/serverController.js [hunk 37]
```diff
@@ -247,12 +257,10 @@ class ServerController extends BaseController {
 
         // resolve associated http route for each actions
         const routeDescriptionList = httpRoutes.filter(route => {
-          return (
-            (route.controller === controller
+          return ((
```
Comment (kuzzle) [inline lib/api/controllers/serverController.js]: MINOR Remove those useless parentheses. [rule](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AParentheses)

**Comment by kuzzle** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
